src/app.js

```diff
@@ -1,41 +1,101 @@
-var auth = require('./resources/authentication');
-var vm = require('./resources/vms');
+var request = require('request')
 
-
-function findVMByName(vmName) {
-  return vm.find('filter.names.1=' + vmName).then(resp => {
-     if (null !== resp && null !== resp.body.value && resp.body.value.length > 0) {
-       return resp.body.value[0].vm;
-     }
-     throw "VM not found"
- });
+async function getDC (action, settings) {
+  let host = action.params.host || settings.host
+  let connect = await initialConnect(action, settings);
+  const token = JSON.parse(connect.body).value
+  const cookie = `vmware-api-session-id=${token}`
```

Comment:
This all should be in `initialConnect`.
It should be changed to `getCookie` and return the final cookie so you don't need these 4 lines to repeat on every function